content/automate/ManageIQ/System/Request.class/__methods__/vm_retire_extend.rb

```diff
@@ -0,0 +1,63 @@
+#
+# Description: This method is used to add 14 days to retirement date when target
+# VM has a retires_on value and is not already retired
+#
+
+module ManageIQ
+  module Automate
+    module System
+      module Request
+        class VmRetireExtend
+          def initialize(handle = $evm)
+            @handle = handle
+          end
+
+          def main
+            validate
+            extend_retirement
+          end
+
+          private
+
+          def vm
+            @vm ||= @handle.root["vm"].tap do |vm|
+              if vm.nil?
+                @handle.log(:error, 'vm is nil')
+                raise 'ERROR - vm object not passed in'
+              end
+            end
+          end
+
+          def validate
+            if vm.retires_on.blank?
+              raise "ERROR - VM #{vm.name} has no retirement date - extension bypassed. No Action taken"
+            end
+
+            if vm.retired
+              raise "ERROR - VM #{vm.name} is already retired - extension bypassed. No Action taken"
+            end
+          end
+
+          def retirement_extend_days
+            @vm_retire_extend_days ||= @handle.object['vm_retire_extend_days'] || 14
```

Comment:
@billfitzgerald0120 
Can you add a constant for 14 at the top of the class 

DEFAULT_EXTENSION_DAYS = 14


And then in the method you could use
``
@vm_retire_extend_days ||= @handle.object['vm_retire_extend_days'] || DEFAULT_EXTENSION_DAYS
```